sdk/core/azure-core/azure/core/pipeline/policies/_retry.py

```diff
@@ -317,7 +318,25 @@ def increment(self, settings, response=None, error=None):
                 settings['status'] -= 1
                 settings['history'].append(RequestHistory(response.http_request, http_response=response.http_response))
 
-        return not self.is_exhausted(settings)
+        if not self.is_exhausted(settings):
+            if response.http_request.body and hasattr(response.http_request.body, 'read'):
+                return self.is_seekable(response.http_request.body)
+            return True
+        return False
```

Comment:
Last thing I might offer, I do personally and this style (with less nesting but earlier return) easier to reason about:
```
if self.is_exhausted(settings):
    return False

if response.http_request.body and hasattr(response.http_request.body, 'read'):
    return self.is_seekable(response.http_request.body)
return True
```

Reply:
Yes. I like it too. Let's make it.